libnopegl/src/glsl/helper_noise.glsl

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2023 Matthieu Bouron <matthieu.bouron@gmail.com>
+ * Copyright 2023 Clément Bœsch <u pkh.me>
+ * Copyright 2023 Nope Foundry
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Return a random single-precision float between [0;1)
+ * Derived from http://prng.di.unimi.it/
+ */
+float u32tof32(highp uint x)
+{
+    uint y = (0x7FU << 23U) | (x >> 9U);
+    return uintBitsToFloat(y) - 1.0;
+}
+
+/*
+ * lowbias32 hashing from https://nullprogram.com/blog/2018/07/31/
+ */
+highp uint lowbias32(highp uint x)
+{
+    x ^= x >> 16;
+    x *= 0x7feb352dU;
+    x ^= x >> 15;
+    x *= 0x846ca68bU;
+    x ^= x >> 16;
+    return x;
+}
+
+uint hash(uvec2 x)
```

Comment:
shouldn't `uvec2` and the returned `uint` be `highp`, just like `lowbias32`? All these functions are pretty much the same (you may want to rename `lowbias32` to `hash()` as well for consistency btw)

Reply:
Added `highp` precision on hash and rand functions.